pandas/core/dtypes/common.py

```diff
@@ -1591,6 +1591,13 @@ def infer_dtype_from_object(dtype) -> type:
     return infer_dtype_from_object(np.dtype(dtype))
 
 
+def holds_integer(value) -> bool:
+    """
+    Whether the type is an integer type.
+    """
+    return lib.infer_dtype(value) in ["integer", "mixed-integer"]
```

Comment:
@topper-123 relying on this at all for internal purposes is what i dislike.  whether its a method vs function is totally irrelevant

Reply:
Ok, can you explan what's the issue with this approach? This method was only used in the plotting module, so a very little used function? Should we replace it there with normal dtype checking, i.e. just drop the "mixed-integer" checks?

Reply:
this is "values-dependent behavior" which in general we are trying to move away from.  it makes it much harder to reason about code.  one of the reasons this is little-used is i have been trying to remove usages of it.

i expect most of the relevant cases could just be `is_integer_dtype(self.dtype)`.  there would likely be some straggler object-dtype-of-integer cases that would need some deprecation cycle